aptos-move/framework/src/natives/hash.rs

```diff
@@ -30,37 +30,40 @@ pub struct SipHashGasParameters {
 /// Serialize the MoveValue with BCS and then feed the bytes into SipHasher. This is not
 /// cryptographically secure.
 fn native_sip_hash(
-    _gas_params: &SipHashGasParameters,
+    gas_params: &SipHashGasParameters,
     context: &mut NativeContext,
     mut ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
     debug_assert!(ty_args.len() == 1);
     debug_assert!(args.len() == 1);
 
-    // TODO(Gas): proper gas metering
+    let mut cost = gas_params.base_cost;
 
     let ref_to_val = pop_arg!(args, Reference);
     let arg_type = ty_args.pop().unwrap();
 
     // delegate to the BCS serialization for `Value`
+    // TODO: I'm confused: what cost will this return with the ? operator
```

Comment:
this is an internal error I believe will fail the whole execution (no gas would be charged)

Reply:
@vgao1996, this doesn't sound right, I assume? We always want to charge gas, no?

Reply:
Anyway, I'm about to change to do the BCS stuff in Move, so these ? operator calls might disappear anyway... Let me see.

Reply:
Zekun was right on this. This function is supposed to never fail, so if it really happens it's an internal invariant violation. Consider it as "gentler VM crash", where we stop to care about gas metering for the current transaction.

Reply:
Okay. Well, it's fixed now after, ahem, ahem, _Moving_ the BCS stuff :)